test/river/request_test.exs

```diff
@@ -6,6 +6,48 @@ defmodule River.RequestTest do
     { :ok, %{request: %Request{uri: URI.parse("https://google.com")}} }
   end
 
+  describe "Request.new/4" do
```

Comment:
I also nested these tests in a describe block. The rest of the tests don't follow this pattern but it's something I'd like to propose. Again if you're against it I can unnest these :D.